Set locale to LC_ALL in Quickstart script. Fixes #2043

lucasplus opened the pull request:
I know #2043 is technically closed. Please refer to the issue page and
see my comment (made just now) showing why it really wasn't fixed.

avaris commented:
`Python >= 2` is essentially all Python versions (we don't support `Python 1` :)). In that case, it's better to not have an `if`.

lucasplus commented:
I agree with your comment @avaris. Based on [this,](https://docs.python.org/3.6/library/locale.html) removing the if statement should be fine.

justinmayer commented:
@avaris: Are there any disadvantages to always setting locale to LC_ALL in the quickstart script for all users?

avaris commented:
Looking at it a bit more in depth...

I think it's fine. Alternatively we can use `getdefaultlocale()` instead of `getlocale()`. That should behave better. Still, according to the docs, `None` may be returned regardless. I think it is better to handle that as well.

lucasplus commented:
About `getdefaultlocale()` , it returns something different on my system than `getlocale()`

So. It has to be `getlocale()`. Right?

In the case of `None`, should we have a default language? English? 

```
>>> locale.getlocale()
('English_United States', '1252')
>>> locale.getlocale()[0].split('_')[0]
'English'
>>> locale.getdefaultlocale()
('en_US', 'cp1252')
>>> locale.getdefaultlocale()[0].split('_')[0]
'en'
```

justinmayer commented:
Any thoughts on the latest changes here, @avaris?

avaris commented on pelican/tools/pelican_quickstart.py:
```diff
@@ -21,8 +21,12 @@
-if (sys.version_info.major == 2):
-    locale.setlocale(locale.LC_ALL, '')
+locale.setlocale(locale.LC_ALL, '')
+language = locale.getlocale()[0]
```
For consistency, can we name the variable like the others (`_DEFAULT_TIMEZONE`)? Something like `_DEFAULT_LANGUAGE`. Otherwise, looks good.

lucasplus commented:
@avaris Thanks for the suggestion. Let me know if you think of anything else.

justinmayer commented:
Looks good, Lucas. Would you please [rebase & squash your commits](https://github.com/getpelican/pelican/wiki/Git-Tips#squashing-commits) so we can get this merged?

lucasplus commented:
@justinmayer I was just reading about rebasing and squashing. I have not done those operations before. I see that @avaris just approved all the commits in this pull request. Should I still rebase and squash? Are you up for getting a question on how to do that? I haven't quite figured it out yet.

justinmayer commented:
> Should I still rebase and squash?

Yes, please.

> Are you up for getting a question on how to do that? I haven't quite figured it out yet.

Have you reviewed the instructions in the link I provided above? If that's not clear, please [visit the #pelican IRC channel](http://docs.getpelican.com/en/stable/contribute.html#the-pelican-irc-channel), and someone will likely be around to assist you. Just be sure to stay connected for a while to ensure you'll see the (eventual) reply.

lucasplus commented:
Ok. I think I did it right. Please let me know if I did not.

I had read your link. I was confused about what to put for `upstream` and `upstream/master`. I believe I got it. 

Thanks for the IRC reference. That will be useful to know.

justinmayer commented:
Many thanks to @lucasplus for the fix and to @avaris for reviewing!  (^_^)

justinmayer commented:
Pelican 3.7.1 has been released, containing the fix from this pull request.
